src/mds/MDSRank.cc

```diff
@@ -3192,11 +3192,21 @@ bool MDSRank::evict_client(int64_t session_id,
     return false;
   }
 
+  auto& addr = session->info.inst.addr;
+  {
+    CachedStackStringStream _ss;
+    auto& ss = _ss.get_stream();
```

Comment:
this code should be rarely executed. do we need this optimization

Reply:
Creating an iostream is expensive. The optimization is not necessary here but I'd like to see increased use of CachedStackStringStream by having it appear in more code.